src/coreclr/gc/gc.cpp

```diff
@@ -6991,7 +6991,8 @@ void gc_heap::gc_thread_function ()
 
     while (1)
     {
-        assert (!gc_t_join.joined());
+        // inactive GC threads may observe gc_t_join.joined() being true here
+        assert ((n_heaps <= heap_number) || !gc_t_join.joined());
```

Comment:
how can joined be true for the inactive GC threads?

Reply:
`gc_t_join.joined()` just returns the value of a static field that will flicker based on what the active GC threads are doing, so occasionally this will fire for the inactive threads.